nx/lib/nx/defn/compiler.ex

```diff
@@ -268,11 +268,14 @@ defmodule Nx.Defn.Compiler do
     res
   end
 
-  defp get_and_normalize_definition(def, state) do
+  defp get_and_normalize_definition(def, function_type, state) do
     {:v1, kind, meta, clauses} = Module.get_definition(state.module, def)
     state = %{state | function: def, line: meta[:line] || state.line, rewrite_underscore?: true}
 
     case clauses do
+      [{meta, args, [], ast}] when function_type == :transform ->
+        {{kind, meta, args, ast}, state}
```

Comment:
I think we can relax the conditions for when `function_type == :transform` here, right? Like, allow multiple clauses, guards and so on?

Reply:
I would not allow it for now, I would keep it as restricted as `defn`. You may need to execute those lines on `args` and `ast` though:

https://github.com/elixir-nx/nx/blob/863ea5a6040dc6d620aabc80fd62b07df1ae5b31/nx/lib/nx/defn/compiler.ex#L427-L430

Reply:
Is there a test I could add where things would break without this `prewalk`?